packages/wallet/src/services/DelegationTracker/RewardsHistory.ts

```diff
@@ -34,11 +34,20 @@ export const createRewardsHistoryProvider =
 
 export type RewardsHistoryProvider = ReturnType<typeof createRewardsHistoryProvider>;
 
-const firstDelegationEpoch$ = (transactions$: Observable<TxWithEpoch[]>) =>
+const firstDelegationEpoch$ = (transactions$: Observable<TxWithEpoch[]>, rewardAccounts: Cardano.RewardAccount[]) =>
   transactions$.pipe(
     map((transactions) =>
       first(
-        transactions.filter(({ tx }) => transactionHasAnyCertificate(tx, [Cardano.CertificateType.StakeDelegation]))
+        transactions.filter(({ tx }) => {
+          const hasCertificate = transactionHasAnyCertificate(tx, [Cardano.CertificateType.StakeDelegation]);
+          if (hasCertificate) {
+            const inspectTx = createTxInspector({
+              sentInspection: sentInspector({ rewardAccounts })
```

Comment:
Maybe let's hoist the sentInspector part that finds certificates into a separate inspector